### ISSUE TITLE
RichText ignores system font size (accessibility)

### ISSUE THREAD

**Issue by AndroidNils** [flutter/flutter#61802]
## Description
`RichText` has no accessibility support. If users change the system font size, the `RichText` widget doesn't change the font size.

## Demo
![RichTextFont](https://user-images.githubusercontent.com/24459435/87862161-62794180-c94d-11ea-8646-accf8b0f78a0.gif)

## Steps to reproduce
<details>
<summary>Code to reproduce</summary>

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Demo',
      home: Scaffold(
        body: SafeArea(
          child: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                RichText(
                  text: TextSpan(
                    children: <TextSpan>[
                      TextSpan(text: "RichText"),
                    ],
                    style: TextStyle(color: Colors.grey[600]),
                  ),
                ),
                Text("Text")
              ],
            ),
          ),
        ),
      ),
    );
  }
}

```
</details>

1. Copy code
2. Run app
3. Change system font size

---

Its working with `textScaleFactor: MediaQuery.of(context).textScaleFactor`. Why isn't this the default value?
```
RichText(
  text: TextSpan(
    children: <TextSpan>[TextSpan(text: "RichText")],
  ),
  textScaleFactor: MediaQuery.of(context).textScaleFactor,
),
```

**Comment by jonahwilliams** [comment]
Please see https://github.com/flutter/flutter/issues/61452#issuecomment-658219360

TLDR: We made a mistake naming RichText, it should be called RawText - aka only use this if you want to implement all parts of text layout/sizing/accessibility/theming. Applications should generally use Text.rich
